test/models/UserAnswersHelperSpec.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import base.SpecBase
+import pages.{ContactTelephoneNumberPage, DoYouHaveUTRPage}
+
+class UserAnswersHelperSpec extends SpecBase {
+
+  "updateUserAnswersIfValueChanged" - {
+    "must leave UserAnswers unchanged if the value does not change" in {
+      val userAnswers = UserAnswers(userAnswersId)
+        .set(DoYouHaveUTRPage, true).success.value
+        .set(ContactTelephoneNumberPage, "07540000000").success.value
+
+      val result = UserAnswersHelper.updateUserAnswersIfValueChanged(userAnswers, DoYouHaveUTRPage, true)
+
+      whenReady(result) {
+          ua =>
+            ua.get(ContactTelephoneNumberPage) mustBe Some("07540000000")
+      }
```

Comment:
I'ld expect this to be a mock of useranswers with a verify that the set method hasn't been called, also just generate the telephone number or use a non-real "telephone" or something.